src/org/ods/util/HelmStatusSimpleData.groovy

```diff
@@ -0,0 +1,186 @@
+     */
+    Map<String, List<String> > resourcesByKind
+
+    @SuppressWarnings(['Instanceof'])
+    @NonCPS
+        static HelmStatusSimpleData fromJsonObject(Object object) {
+        try {
+            def rootObject = ensureMap(object, "")
+
+            def infoObject = ensureMap(rootObject.info, "info")
+
+            // validation of missing keys or types
+            def missingKeys = []
+            def badTypes = []
+            def expectedStringAttribsForRoot = collectMissingStringAttributes(rootObject,
+                ["name", "namespace"])
+            def expectedStringAttribsForInfo = collectMissingStringAttributes(infoObject,
+                ["status", "description", "last_deployed"])
+            missingKeys.addAll(expectedStringAttribsForRoot.first)
+            missingKeys.addAll(expectedStringAttribsForInfo.first)
+            badTypes.addAll(expectedStringAttribsForRoot.second)
+            badTypes.addAll(expectedStringAttribsForInfo.second)
+            def att = "version"
+            if (!rootObject.containsKey(att)) {
+                missingKeys << att
+            } else if (!(rootObject[att] instanceof Integer)) {
+                badTypes << "${att}: expected Integer, found ${rootObject[att].getClass()}"
+            }
+            handleMissingKeysOrBadTypes(missingKeys, badTypes)
+            def resourcesObject = ensureMap(infoObject.resources, "info.resources")
+            // All resources are in an json object which organize resources by keys
+            // Examples are  "v1/Cluster", "v1/ConfigMap"  "v1/Deployment", "v1/Pod(related)"
+            // For these keys the map contains list of resources.
+            Map<String, List<String>> resourcesByKind = [:] .withDefault { [] }
+            for (entry in resourcesObject.entrySet()) {
+                def key = entry.key as String
+                def resourceList = ensureList(entry.value, "info.resources.${key}")
+                resourceList.eachWithIndex { resourceJsonObject, i ->
+                    def resourceContext = "info.resources.${key}.[${i}]"
+                    def resource = extractResource(resourceJsonObject, resourceContext)
+                    if (resource) {
+                        resourcesByKind[resource.first] << resource.second
+                    }
+                }
+            }
+            def hs = new HelmStatusSimpleData()
+            hs.with {
+                name = rootObject.name
+                version = rootObject.version as String
+                namespace = rootObject.namespace
+                status = infoObject.status
+                description = infoObject.description
+                lastDeployed = infoObject["last_deployed"]
+            }
+            hs.resourcesByKind = resourcesByKind.collectEntries { kind, names ->
+                [ kind, names.collect() ]
+            } as Map<String, List<String> >
+            return hs
+        } catch (IllegalArgumentException ex) {
```

Comment:
Just let this exception propagate. Illegal argument exception shouldn't normally happen and it contains enough information for debug.

Reply:
This catches the exception with a report message, as metioned on: https://github.com/opendevstack/ods-jenkins-shared-library/pull/1143#discussion_r1806269048